prep.py

```diff
@@ -99,6 +92,10 @@ def accounts_json(num_files, n, k):
 def create_weather(growth=32):
     filenames = sorted(glob(os.path.join('data', 'weather-small', '*.hdf5')))
 
+    if not filenames:
+            ws_dir = os.path.join('data', 'weather-small')
```

Comment:
Should it not be `data_dir`? It is conceivable that the user runs this file while not in the right directory.

Reply:
Keeping the check as before and creating a meaningful error sounds very reasonable. I think it should be safe to remove the weather-small check in that case. Otherwise, using `data_dir` instead does sound preferable.

Should we provide additional instructions on how to fix the issue within the error? "Reclone the repository"? "Create data/ and download weather-small from the repo"?

Reply:
> I think it should be safe to remove the weather-small check in that case. Otherwise, using data_dir instead does sound preferable.

Adding a check that files in `weather-small` are found is still useful (using `data_dir` instead of `'data'`). It's possible that the `.hdf5` files could have been deleted for some reason. 

> Should we provide additional instructions on how to fix the issue within the error? 

That's a good idea! I think something like "data/ directory not found, aborting data preparation. Please re-clone the dask-tutorial repository." would be sufficient instructions.

Reply:
I made those changes as suggested.

>Should it not be data_dir? It is conceivable that the user runs this file while not in the right directory.

I am noticing there are other places in the code that use `'data'` instead of `data_dir`. Would you like me to change those to `data_dir`, probably in a different commit?

Reply:
> Would you like me to change those to data_dir

Yes, please! Also, it should have `os.path.asbpath()` around the definition of data_dir :)

Reply:
Should I include that with this PR or create a new one? It sounds like a separate issue to me, but I'll do whatever you'd like.

Reply:
Either one is fine, whichever you prefer